### PR TITLE
Reduce bmpslot usage by headz

### PR DESCRIPTION
**Pull request by niffiwan**
As headani's (both ANI & EFF) are streamed there's no need to have
the anim data in bmpman, the streaming code does that with only two
slots, rather than one slot per ani frame
Also fix a "leak" in the streaming code use of bmpman slots. i.e. each
time a message was played another slot would be used & not released

### CODE REVIEW

File: code/mission/missionmessage.cpp [hunk 1]
```diff
@@ -316,7 +316,8 @@ int add_avi( char *avi_name )
 	strcpy_s( extra.name, avi_name );
 	strcpy_s( extra.anim_data.filename, avi_name);
 	extra.num = -1;
-	generic_anim_load(&extra.anim_data);
+	generic_anim_load(&extra.anim_data);   // load only to validate the anim
+	generic_anim_unload(&extra.anim_data); // unload to not waste bmpman slots
```
Comment (asarium) [inline code/mission/missionmessage.cpp]: Doesn't this reset all the data in `extra.anim_data`? If that's the case, what's the reason to have those two load/unload calls in the first place if the data is not going to be used?

Comment (niffiwan) [inline code/mission/missionmessage.cpp]: The only reason to load the animation data is to validate the animation being used, i.e. that EFFs have all their frames, they have a valid image subtype, etc. The alternative would be to remove both the load/unload (i.e. return it to retail behaviour) however I thought there was a benefit to validating the animations here rather than potential issues being found in-mission.

Comment (asarium) [inline code/mission/missionmessage.cpp]: But how is it validated? `generic_anim_load` has an error return value but it is never checked so no matter if the animation is invalid or not no warning will be displayed or am I missing something obvious here?

Comment (niffiwan) [inline code/mission/missionmessage.cpp]: Ah, right. `generic_anim_load` calls `bm_load_animation` and that has several Warnings for bad data. 
(Note that retail ANIs will always fail to load here due to [abc] being appended to the tabled filename to find the real files, so checking the return code would require exceptions).

Comment (asarium) [inline code/mission/missionmessage.cpp]: Does the current code display a warning if an EFF head ani doesn't exist? As far as I can tell it doesn't but that's a pretty important part of validating the table entry.

Comment (niffiwan) [inline code/mission/missionmessage.cpp]: I should confirm this with test data, but I think you're right. There'll only be this message from `read_raw_file_text` if the EFF file doesn't exist:

   nprintf(("Error", "Wokka!  Error opening file (%s)!\n", filename));

Comment (asarium) [inline code/mission/missionmessage.cpp]: Yeah, I agree. Better modder feedback would be nice but shouldn't be part of this PR.

**Comment by asarium** [comment]
LGTM
